pkg/scheduler/test_scheduler.go

```diff
@@ -411,3 +414,29 @@ func mapK8sImagePullSecrets(secrets []v1.LocalObjectReference) []string {
 	}
 	return res
 }
+
+func mergeArtifacts(artifactBase *testkube.ArtifactRequest, artifactAdjust *testkube.ArtifactRequest) *testkube.ArtifactRequest {
+	if artifactBase == nil && artifactAdjust == nil {
+		return nil
+	}
+
+	if artifactBase == nil && artifactAdjust != nil {
+		return artifactAdjust
+	}
+
+	if artifactBase != nil && artifactAdjust == nil {
+		return artifactBase
+	}
+
+	if artifactBase.VolumeName == "" && artifactAdjust.VolumeName != "" {
+		artifactBase.VolumeName = artifactAdjust.VolumeName
+	}
+
+	if artifactBase.VolumeMountPath == "" && artifactAdjust.VolumeMountPath != "" {
+		artifactBase.VolumeMountPath = artifactAdjust.VolumeMountPath
+	}
```

Comment:
all these ifs could be shortened using a switch..case with empty statement in switch, ex:
```
switch {
case base == nil && adjust == nil:
return nil
case base != nil && adjust == nil:
return base
...
}
```

Reply:
hope, it's a bit better now